### DISCUSSION TITLE
Is it possible to extend default interpolated-string handler to accept string.Join without an intermediate string allocation

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/runtime#61546] [Q&A]
Is there a way to avoid this kind of code to be emitted?
```cs
defaultInterpolatedStringHandler.AppendFormatted(string.Join(' ', a));
```

**Comment by stephentoub** [comment]
I'm not following.  What C# code is generating that?

**Reply by alrz** [reply]
For example: `$"abc{1}{string.Join(' ', a)}"`

It's probably not possible to do this without changing the code, but I'm wondering if anything could be done here, maybe using a custom handler?

**Reply by stephentoub** [reply]
> I'm wondering if anything could be done here, maybe using a custom handler

No,  this would require roslyn to recognize string.Join and emulate its semantics by generating equivalent code.
